dataflow/gemma-flex-template/custom_model_gemma.py

```diff
@@ -0,0 +1,179 @@
+        """
+        model_config = get_config_for_2b(
+        ) if "2b" in model_variant else get_config_for_7b()
+        model_config.tokenizer = tokenizer_path
+        model_config.quant = 'quant' in model_variant
+        model_config.tokenizer = tokenizer_path
+
+        self._model_config = model_config
+        self._checkpoint_path = checkpoint_path
+        if device == 'GPU':
+            logging.info("Device is set to CUDA")
+            self._device = torch.device('cuda')
+        else:
+            logging.info("Device is set to CPU")
+            self._device = torch.device('cpu')
+        self._env_vars = {}
+
+    def share_model_across_processes(self) -> bool:
+        return True
+
+    def load_model(self) -> GemmaForCausalLM:
+        """Loads and initializes a model for processing."""
+        torch.set_default_dtype(self._model_config.get_dtype())
+        model = GemmaForCausalLM(self._model_config)
+        model.load_weights(self._checkpoint_path)
+        model = model.to(self._device).eval()
+        return model
+
+    def run_inference(
+        self,
+        batch: Sequence[str],
+        model: GemmaForCausalLM,
+        inference_args: Optional[Dict[str, Any]] = None
+    ) -> Iterable[PredictionResult]:
+        """Runs inferences on a batch of text strings.
+
+        Args:
+          batch: A sequence of examples as text strings.
+          model: The Gemma model being used.
+          inference_args: Any additional arguments for an inference.
+
+        Returns:
+          An Iterable of type PredictionResult.
+        """
+        # Loop each text string, and use a tuple to store the inference results.
+        predictions = []
+        result = model.generate(prompts=batch, device=self._device)
+        predictions.append(result)
+        return [PredictionResult(x, y) for x, y in zip(batch, predictions)]
+
+
+class FormatOutput(beam.DoFn):
+    def process(self, element, *args, **kwargs):
+        yield "Input: {input}, Output: {output}".format(
+            input=element.example, output=element.inference)
+
+
+if __name__ == "__main__":
+    import argparse
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--messages_subscription",
+        required=True,
+        help="Pub/Sub subscription for input text messages",
+    )
+    parser.add_argument(
+        "--responses_topic",
+        required=True,
+        help="Pub/Sub topic for output text responses",
+    )
+    parser.add_argument(
+        "--model_variant",
+        required=False,
+        default="gemma-2b-it",
+        help="name of the gemma variant being used",
+    )
+    parser.add_argument(
+        "--checkpoint_path",
+        required=False,
+        default="pytorch_model/gemma-2b-it.ckpt",
+        help="path to the Gemma model weights in the custom worker container",
+    )
+    parser.add_argument(
+        "--tokenizer_path",
+        required=False,
+        default="pytorch_model/tokenizer.model",
```

Comment:
is this part of gemma distribution?

Reply:
Yes the tokenizer is included when you download the pytorch version of gemma